### PR TITLE
Kserve upgrade

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:

This PR contains various changes but mainly concerned on:
1. Upgrading kserve to version 0.8 ( https://github.com/gojek/merlin/pull/221)
2. Introducing native / raw deployment ( https://github.com/gojek/merlin/pull/227)
3. Introducing configurable autoscaling strategy (https://github.com/gojek/merlin/pull/229)
4. Improvement in the CI and end to end test

**Which issue(s) this PR fixes**:

None

**Does this PR introduce a user-facing change?**:

```release-note
- Add deployment mode configuration
- Implement configurable autoscaling strategy
```

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [x] Updated documentation
- [x] Update Swagger spec if the PR introduce API changes
- [x] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: api/service/version_endpoint_service.go [hunk 1]
```diff
@@ -172,12 +185,13 @@ func (k *endpointService) DeployEndpoint(environment *models.Environment, model
...
+	} else {
 		endpoint.EnvVars = newEndpoint.EnvVars
 	}
 
-	originalEndpoint := *endpoint
-
+	endpoint.Status = models.EndpointPending
+	// Copy to avoid race condition
+	tobeDeployedEndpoint := *endpoint
 	endpoint.Status = models.EndpointPending
```
Comment (tiopramayudi) [inline api/service/version_endpoint_service.go:195]: why set this twice ya?

Comment (pradithya) [reply api/service/version_endpoint_service.go:195]: I think it's typo when resolving some merge conflict

File: .github/workflows/merlin.yml [hunk 2]
```diff
@@ -0,0 +1,438 @@
+name: Merlin CI Workflow
+on:
+  push:
+    branches: 
+    - main 
+    - kserve_upgrade
```
Comment (ariefrahmansyah) [inline .github/workflows/merlin.yml:6]: TODO: Should remove kserve_upgrade later

File: api/service/version_endpoint_service.go (hunk 1 above)
Comment (pradithya) [reply api/service/version_endpoint_service.go:194]: Updated

File: Makefile [hunk 3]
```diff
@@ -173,10 +190,42 @@ generate-client-python:
...
+
+# ============================================================
+# Docker build
+# ============================================================
+.PHONY: docker-build
+docker-build: docker-build-transformer docker-build-api docker-build-pyfunc docker-build-batch-predictor
+
+.PHONY: docker-build-api
+docker-build-api: build-ui
+	@cp -r ui/build build
```
[the remark this answers is not in the document]
Comment (pradithya) [reply Makefile]: Updated

**Review by ariefrahmansyah — APPROVED** [review]

**Pull request merged**